qf/submission.go

```diff
@@ -14,10 +14,10 @@ func (s *Submission) IsApproved() bool {
 // NewestSubmissionDate returns the submission's build date if newer than the provided submission date.
 // Otherwise, the provided submission date is returned, i.e., if it is newer.
 func (s *Submission) NewestSubmissionDate(submissionDate time.Time) (t time.Time, err error) {
-	if s == nil || s.BuildInfo == nil {
+	if s == nil || s.BuildInfo == nil || s.BuildInfo.BuildDate == nil || s.BuildInfo.SubmissionDate == nil {
 		return t, ErrMissingBuildInfo
 	}
-	currentSubmissionDate, err := time.Parse(TimeLayout, s.BuildInfo.SubmissionDate)
+	currentSubmissionDate := s.BuildInfo.SubmissionDate.AsTime()
```

Comment:
If instead we use the getter methods for build info and submission date I think we maybe don’t need the nil checking above?

Reply:
We can totally skip the nil checking. I assumed the point was to return with an error immediately in case `BuildInfo` is missing.

Reply:
This is what I meant:
```go
	currentSubmissionDate := s.GetBuildInfo().GetSubmissionDate().AsTime()
	if currentSubmissionDate.IsZero() {
		return t, ErrMissingBuildInfo
	}
```
This is possible since even `s == nil` will work fine since `GetBuildInfo()` checks for this and returns `nil` and so on. All the way to `AsTime()`, which will return 0 for each component of `timestamppb.Timestamp` instead of `nil`, which should result in returning the zero time.

PS: I didn't test the code, so please add a test for this... to be sure.

Reply:
I fixed this by removing the need for even returning an error. Leaving the comment unresolved FYI.